### PR TITLE
Deprecate H5DataIO from videointerface

### PR DESCRIPTION
**Pull request by h-mayorquin**
Draft

### CODE REVIEW

File: src/neuroconv/datainterfaces/behavior/video/videodatainterface.py [hunk 1]
```diff
@@ -437,14 +436,18 @@ def add_to_nwbfile(
...
-                wrapped_io_data = H5DataIO(
-                    iterable,
-                    compression=compression,
-                    compression_opts=compression_options,
-                    chunks=chunks,
+            # TODO: remove completely after 03/1/2024
+            if compression is not None or compression_options is not None:
+                warnings.warn(
+                    message=(
+                        "Specifying compression methods and their options at the level of tool functions has been deprecated. "
```
Comment (CodyCBakerPhD) [inline src/neuroconv/datainterfaces/behavior/video/videodatainterface.py:443]:
```suggestion
                        "Specifying compression methods and their options for this interface has been deprecated. "
```

Copied and pasted from elsewhere? This isn't a tool function, but an interface

Comment (h-mayorquin) [reply src/neuroconv/datainterfaces/behavior/video/videodatainterface.py:443]: Yes, I did, thanks for the catch!

**Comment by CodyCBakerPhD** [comment]
One comment on warning

Can you also remove the docstring for these arguments? https://github.com/catalystneuro/neuroconv/blob/cade1204bb50d08cd03dd887d38ea61eb3f6e1ad/src/neuroconv/datainterfaces/behavior/video/videodatainterface.py#L323-L329

What do you think about the tests in https://github.com/catalystneuro/neuroconv/blob/b9507bf58a977f90f03aed743c272c059cc21fbf/tests/test_behavior/test_video_utils.py#L154 - necessary to keep or can be removed?

**Comment by h-mayorquin** [comment]
I removed the docstrings. 

Concerning the tests I am leaning to remove them just to make the testing slightly simpler but I really don't feel strongly either way.

**Comment by CodyCBakerPhD** [comment]
I agree with removing; it isolates the responsibility of that support to the main configuration tool testing

**Comment by h-mayorquin** [comment]
OK, removed.

**Review by CodyCBakerPhD — APPROVED** [review]

**Comment by CodyCBakerPhD** [comment]
Cool!

Next step is to finally unlock the Zarr option in places like https://github.com/catalystneuro/neuroconv/blob/main/src/neuroconv/nwbconverter.py#L175 (by including it in the literal)

Together with enhancing testing to ensure all interfaces can write+read back Zarr format

Did you want to handle that too?

**Comment by h-mayorquin** [comment]
Yep. Let's do it.

**Pull request merged**